add chapter documenting the HPC-UGent web portal (Open OnDemand) (HPC-8496) (#380)
hpcugent/vsc_user_docs: intro-HPC/ch_web_portal.tex

\chapter{Using the HPC-UGent web portal}
\label{ch:web_portal}

> \href{https://openondemand.org/}{\strong{Open OnDemand}} (or \texttt{OoD} for short)

Reviewer: from ohio docs: 
```
We provide a number of web portals for our user community.

OSC OnDemand is our "one stop shop" for access to our High Performance Computing resources. With OnDemand, you can upload and download files, create, edit, submit, and monitor jobs, run GUI applications, and connect via SSH, all via a web broswer, with no client software to install and configure. Full documentation is available. 

```

to get inspired :wink:

Author: I got inspired.

Reviewer: I would only mention OOD somewhere that the portal is based on it, but keep the terminology on "webportal". this is not relevant in the explanation.

Reviewer: eg ` The webportal provides a one stop shop ...`

Author: fixed in 119b9af, Open OnDemand is now only mention twice: to mention the web portal is based on it, and when linking to OoD docs.

is a web portal to access HPC resources.

You can use it to connect to the HPC-UGent infrastructure via:

\begin{center}\url{https://login.hpc.ugent.be}\end{center}

You do not need to install any additional software to use the web portal, a standard web browser like Firefox,

> Google Chrome or Safari (macOS) is sufficient. In addition, no SSH key pair is required to connect

Reviewer 2: ```suggestion
Google Chrome or Safari (MacOS) is sufficient. In addition, no SSH key pair is required to connect
```

Author: no longer relevant (removed)

to your VSC account.

Through the web portal, you can:
\begin{itemize}

>     \item browse through your files \& directories, and inspect, manage or change them;

Reviewer: i would add urls here (or make these "clickable")

Author: There's no general URL to open the file browser to your home directory, the URL includes your VSC login id.

Also, what if for some reason URLs change in OoD. That'll result in broken links in the docs.

So, I prefer not to add clickable links, the menu is pretty obvious once you're logged in to the portal.

    \item consult active jobs (across all HPC-UGent Tier-2 clusters);
    \item submit new jobs to the HPC-UGent Tier-2 clusters, either from existing job scripts or from job templates;
    \item start an interactive graphical user environment (a.k.a. desktop), either on the login nodes or on a cluster workernode;
    \item open a terminal session directly in your web browser;
\end{itemize}

More detailed information is available below, as well as in the \href{https://osc.github.io/ood-documentation/master/}{Open OnDemand documentation}.

> An Open OnDemand walkthrough video is available on YouTube \href{https://www.youtube.com/watch?v=4-w-4wjlnPk}{here}.

Reviewer 3: :+1:

Author: :)

\section*{Pilot access}

\strong{The HPC-UGent web portal is currently available for all VSC accounts,

> but there may be a couple of small problems that we still need to resolve.}

Reviewer 2: ```suggestion
but there may be a couple of small problems that still need to be resolved.}
```

Author: suggested change made + add "Known issus & limitations" subsection mentioning limited resources.

\strong{If you notice anything that doesn't work as expected or could be improved,
please contact \hpcinfo.}

> \strong{In addition, you \emph{must} be connected to the HPC-UGent network

Reviewer: `you` -> `during pilot phase you`

Author: changed

> \strong{In addition, you \emph{must} be connected to the HPC-UGent network

Reviewer 2: ```suggestion
\strong{In addition, during the OOD pilot phase, you \emph{must} be connected to the HPC-UGent network
```

Reviewer: don't refer  to the portal as OOD please

Author: agreed, I changed this to `In addition, during the pilot phase`

(via \href{https://helpdesk.ugent.be/vpn/en/}{VPN} if necessary)
in order to use the HPC-UGent web portal.}

\section{Login}

When visiting the HPC-UGent web portal via \url{https://login.hpc.ugent.be},

> you will be automatically logged in via the VSC accountpage, which wll in turn

Reviewer 2: ```suggestion
you will be automatically logged in via the VSC accountpage, which will in turn
```

Author: no longer relevant since we're pointing to section 2.2 now, this piece of text was removed (nice catch though!)

> authenticate you via your institutional login service (for UGent also known as CAS).

Reviewer: there is a section that describes accessing the AP already, i would refer to that. i guess most will recognise and click through blindly :smile:

Author: OK, referred to section 2.2

You may need to enter your UGent login name and password in the standard CAS login page if you are not logged in yet.

\subsection{First login}

> The first time you visit \url{https://login.hpc.ugent.be} permission will be requested to let Open OnDemand

Reviewer: `Open OnDemand` -> `web portal`

Author: OK, changed.

> access the VSC accountpage to authenticate you, as shown in this screenshot below:

Reviewer: the authentication part is already passed at this point. this is the user allwoing 3rd party like the webportal access some personal info. in this case, the personal info is the vscid, accoutn status, login shell and insitute name. as it is not dsiplayed in the request, you might point this out here. if needed, i can limit this to the vsc id alone.

Author: OK, clarified.

\begin{center}
    \includegraphics*[scale=0.5]{ood_permission}
\end{center}

\strong{Please click "Authorize" here.}

This request will only be made once, you should not see this again afterwards.

\section{Start page}

Once logged in, you should see this start page:

\begin{center}
    \includegraphics*[scale=0.4]{ood_start}
\end{center}

This page includes a menu bar at the top, with buttons on the left providing access to the different features supported by Open OnDemand,
as well as a \emph{Help} menu, your VSC account name, and a \emph{Log Out} button on the top right,
and the familiar HPC-UGent welcome message with a high-level overview of the HPC-UGent Tier-2 clusters.

> If your browser window is too narrow, the menu is available at the top right through the ``hamburger'' icon:

Reviewer 3: did not know that :+1:

Author: I noticed when I was making my browser window more narrow to make screenshots :P

\begin{center}
    \includegraphics*[scale=0.4]{ood_hamburger}
\end{center}

\section{Features}

> We briefly cover the different features provided by the web portal, going from left to right (or top to bottom) in the menu.

Reviewer: `or top to bottom` is only relevant for the drop down menus?

Author: Yeah indeed, so removed.

This will open a new browser tab providing an interface to create new jobs:

\begin{center}
    \includegraphics*[scale=0.35]{ood_job_composer}
\end{center}

This extensive interface allows you to create jobs from one of the available templates,
or by copying an existing job.

You can carefuly prepare your job and the corresponding job script via the \emph{Job Options} button
and by editing the job script (see lower right).

Don't forget to actually submit your job to the system via the green \emph{Submit} button!

\paragraph{Job templates}

In addition, you can inspect provided job templates, copy them or even create your own templates via the \emph{Templates} button on the top:

\begin{center}
    \includegraphics*[scale=0.35]{ood_job_templates}
\end{center}

\subsection{Shell access}

Through the \emph{Shell Access} button that is available under the \emph{Clusters} menu item,
you can easily open a terminal (shell) session into your VSC account, straight from your browser!

\begin{center}
    \includegraphics*[scale=0.5]{ood_shell}
\end{center}

Using this interface requires being familiar with a Linux shell environment (see Appendix~\ref{ch:useful-linux-commands}).

> To exit the shell session, type \lstinline|exit| followed by \emph{Enter} and then close the browser tab.

Reviewer: @wdpypere @baverhey what happens when a user doesn't exit before exiting the shell. i hope this gets cleaned up on the loginnode

Reviewer: actually, what happens if you reconnect after not exit? do you get to your existing shell, so sort of screen/tmux thingie?

Reviewer 3: @stdweird there is no screen or tmux or anything. if you leave the session or close the tab in any way, it is gone. no reconnecting is possible.

Reviewer 3: @stdweird there is no screen or tmux or whatever. it really works a bit like an actual ssh client. if you close the session or close the tab in any way your connection is broken and cleaned up. no reconnecting possible to the existing shell, it does not exist. the shell app works quite nice.

Author: I clarified this a bit in the text.

\subsection{Interactive applications}

\subsubsection{Graphical desktop environment}

To create a graphical desktop environment, use on of the \emph{desktop on ... node} buttons under \emph{Interactive Apps} menu item. For example:

\begin{center}
    \includegraphics*[scale=0.35]{ood_launch_desktop}
\end{center}

> You can either start a desktop environment on a login node (which should start almost instantly)

Reviewer 3: I would leave the "(which should start almost instantly)" out

Reviewer 4: second that!

Author: dropped the `should start` part + clarified a bit

for some lightweight tasks, or on a workernode of one of the HPC-UGent Tier-2 clusters if
more resources are required (it may take some time before the desktop starts running, depending
on requested resources and how busy the cluster is).

> \strong{Do keep in mind that desktop environments on a cluster workernode are limited a maximum of

Reviewer 2: ```suggestion
\strong{Do keep in mind that desktop environments on a cluster workernode are limited to a maximum of
```

Author: fixed, thx

\chapter{Using the HPC-UGent web portal}
\label{ch:web_portal}

\href{https://openondemand.org/}{\strong{Open OnDemand}} (or \texttt{OoD} for short)
is a web portal that provides ``one stop shop'' for accessing High Performance Computing resources.

> With Open OnDemand, you can upload and download files, create, edit, submit, and monitor jobs,

Reviewer: `with ood` -> `with the portal`

run GUI applications, and connect via SSH, all via a standard web browser like Firefox, Chrome
or Safari. You do not need to install or configure any client software,
and no SSH key is required to connect to your VSC account via this web portal.

> To connect to the HPC-UGent infrastructure via Open OnDemand, visit:

Reviewer: `ood` -> `webportal`

\begin{center}\url{https://login.hpc.ugent.be}\end{center}

> Note that you may only see a ``\emph{Submitting...}'' message appear for a couple of seconds,

Reviewer: hmm, what was this again. i might be able to resolve this or make it more descriptive

Author: Is this a blocker for merging this PR?

It's easy to remove afterwards if it's fixed or made more descriptive...
